### PR TITLE
Fix #16612 - Empty relation view table dropdown

### LINKED ISSUE #16612: Relation view table dropdown empty
5.1.0-rc1
5.2.0-dev

https://demo.phpmyadmin.net/master-config/index.php?route=/table/relation&db=sakila&table=film_actor

image

### PR DESCRIPTION
**Pull request by shucon**
Signed-off-by: Saksham Gupta <shucon01@gmail.com>

### Description

Fixes Empty relation view table dropdown.

Fixes #16612

Before submitting pull request, please review the following checklist:

- [ ] Make sure you have read our [CONTRIBUTING.md](https://github.com/phpmyadmin/phpmyadmin/blob/master/CONTRIBUTING.md) document.
- [ ] Make sure you are making a pull request against the correct branch. For example, for bug fixes in a released version use the corresponding QA branch and for new features use the _master_ branch. If you have a doubt, you can ask as a comment in the bug report or on the mailing list.
- [ ] Every commit has proper `Signed-off-by` line as described in our [DCO](https://github.com/phpmyadmin/phpmyadmin/blob/master/DCO). This ensures that the work you're submitting is your own creation.
- [ ] Every commit has a descriptive commit message.
- [ ] Every commit is needed on its own, if you have just minor fixes to previous commits, you can squash them.
- [ ] Any new functionality is covered by tests.

### CODE REVIEW

**Review by williamdes — CHANGES_REQUESTED** [review]
I tested your fix it does "work" but creates a bug by emptying the table list of any already existing releatin

I would prefer a solution that would fetch the tables in the twig template

**Comment by shucon** [comment]
> I tested your fix it does "work" but creates a bug by emptying the table list of any already existing releatin
> 
> I would prefer a solution that would fetch the tables in the twig template

Thanks for picking up the bug! I've fixed it in twig instead as you suggested.

File: templates/table/relation/foreign_key_row.twig [hunk 1]
```diff
@@ -89,6 +89,8 @@
     {% set tables = [] %}
     {% if foreign_db %}
         {% set tables = get_tables(foreign_db, tbl_storage_engine) %}
+    {% else %}
+        {% set tables = get_tables(db, tbl_storage_engine) %}
```
Comment (williamdes) [inline templates/table/relation/foreign_key_row.twig:93]: db is always defined in this scenario ?

File: templates/table/relation/relational_dropdown.twig [hunk 2]
```diff
@@ -11,7 +11,7 @@
             {{ value }}
         </option>
     {% endfor %}
-    {% if foreign is not same as(false) and not seen_key %}
+    {% if foreign is not same as(false) and foreign != "" and not seen_key %}
```
Comment (williamdes) [inline templates/table/relation/relational_dropdown.twig:14]: could you explain this diff ?

File: templates/table/relation/foreign_key_row.twig (hunk 1 above)
Comment (shucon) [reply templates/table/relation/foreign_key_row.twig:93]: Yes!

File: templates/table/relation/relational_dropdown.twig (hunk 2 above)
Comment (shucon) [reply templates/table/relation/relational_dropdown.twig:14]: If it's not in place, there is an extra empty row that is not required.

Comment (williamdes) [reply templates/table/relation/relational_dropdown.twig:14]: For the record, it will prevent this extra line to exist
image

**Review by williamdes — APPROVED** [review]
It works great !
:100:

**Pull request merged**